### PR TITLE
Fixes #16847: LDAP connection pool error are not correctly reported 

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/16847

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala [hunk 1]
```diff
@@ -231,6 +231,8 @@ object RudderProperties {
 object RudderConfig extends Loggable {
   import RudderProperties.config
 
+  private case class InitError(msg: String) extends Throwable(msg, null, false, false)
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala]: where is it used?

File: webapp/sources/scala-ldap/src/main/scala/com/normation/ldap/sdk/LDAPConnectionProvider.scala [hunk 2]
```diff
@@ -239,8 +239,10 @@ trait PooledConnectionProvider[LDAP <: RoLDAPConnection] extends LDAPConnectionP
     new LDAPConnectionPool(self.newUnboundidConnection, poolSize)
   } catch {
     case ex: LDAPException =>
-      LDAPConnectionLogger.error(s"Error during LDAP connection pool initialisation. Exception: ${ex.getDiagnosticMessage}")
-      throw new Error(ex.getDiagnosticMessage)
+      val msg = s"Error during LDAP connection pool initialisation. Exception: " +
+                s"${ex.getClass.getSimpleName}:${ex.getMessage}; Details: ${ex.getDiagnosticMessage}"
+      LDAPConnectionLogger.error(msg)
+      throw new Error(msg)
```
Comment (ncharles) [inline webapp/sources/scala-ldap/src/main/scala/com/normation/ldap/sdk/LDAPConnectionProvider.scala]: shouldn't it be InitError?

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala [hunk 3]
```diff
@@ -501,11 +503,24 @@ object RudderConfig extends Loggable {
   // such service must not expose implementation details
   //
 
+  // we need that to be the first thing, and take care of Exception so that the error is
+  // human undertandable when the directory is not up
+
+  val roLDAPConnectionProvider: LDAPConnectionProvider[RoLDAPConnection] = roLdap
+  // test connection is up and try to make an human understandable error message.
+  ApplicationLogger.warn(s"LDAP connection active (${ZioRuntime.internal.unsafeRun((for {
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala]: doesn't it mean that we'll get a warning at each start?

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/scala-ldap/src/main/scala/com/normation/ldap/sdk/LDAPConnectionProvider.scala (hunk 2 above)
Comment (fanf) [reply webapp/sources/scala-ldap/src/main/scala/com/normation/ldap/sdk/LDAPConnectionProvider.scala]: no, here we can use `Error`, init error is used in the test for active LDAP connection.

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**